2017/07/02_url_parsing_one.py

```diff
@@ -74,6 +74,43 @@ def url_tuple(url: str) -> Tuple[str, str, str, str, str]:
 
     """
 
+    final_list = []
+
+    if url:
+        protocol_split = url.split('://')
+        protocol = protocol_split[0]
+        working_url = protocol_split[1]
+    else:
+        return '', '', '', '', ''
+
+    if '?' in url:
+        query_split = working_url.split('?')
+        query = query_split[1]
+        working_url = query_split[0]
+    else:
+        query = ''
+
+    if '/' in working_url:
+        path_split = working_url.split('/')
+        path = '/' + path_split[1]
+        contains_host = path_split[0]
+    else:
+        path = ''
+        contains_host = working_url
+
+    if ':' in contains_host:
+        port_split = contains_host.split(':')
+        host = port_split[0]
+        port = port_split[1]
+    else:
+        host = contains_host
+        port = ''
+
+    final_list.extend([protocol, host, port, path, query])
+    return tuple(final_list)
```

Comment:
Rather than extending the list and converting it to a tuple, you could also just return your tuple directly like

```python
return protocol, host, port, path, query
```
or
```python
final = (protocol, host, port, path, query)
return final
```